### ISSUE TITLE
Bug: adding a word to a (large) workspace dictionary causes the workspace dictionary to be deleted

### ISSUE THREAD

**Issue by bradzacher** [streetsidesoftware/vscode-spell-checker#616]
Best shown via a video!

![ezgif-4-0adf620d489a](https://user-images.githubusercontent.com/7462525/100166587-ea2e0a00-2e71-11eb-80c4-0060bea7f373.gif)

[Screen Recording 2020-11-24 at 16.23.32.mov.zip](https://github.com/streetsidesoftware/vscode-spell-checker/files/5593571/Screen.Recording.2020-11-24.at.16.23.32.mov.zip)

<details>
<summary>My `.cspell.json` file</summary>

```json
{
    "version": "0.1",
    "language": "en",
    "ignorePaths": [
        "**/*.json",
        ".cspell.json",
        ".markdownlint.json",
        "yarn.lock",
        ".vscode/*.json",
        "NAMES.md"
    ],
    "dictionaries": [
        "en-gb",
        "misc"
    ],
    "ignoreRegExpList": [
        "/```[\\w\\W]*?```/",
        "/``[\\w\\W]*?``/",
        "/`[^`]*`/"
    ],
    "words": [
        "aashta",
        "adamantite",
        "Aerenal",
        "Aereni",
        "Arawai",
        "Arawai's",
        "Arcanix",
        "Archfey",
        "Argonth",
        "Arrah",
        "Arrah's",
        "Atcha",
        "Aundair",
        "Aundair's",
        "Aundairian",
        "Aundairians",
        "Aundarian",
        "Aurala",
        "Aurala's",
        "Aureon",
        "Awore",
        "Balinor",
        "Banreon",
        "Barmin",
        "Blademarks",
        "Bluevine",
        "Boldrei",
        "Bonal",
        "Bonal's",
        "Boranel",
        "Boranel's",
        "Boromar",
        "Boromar's",
        "Boromars",
        "Borran",
        "Borran's",
        "Brect",
        "Breland",
        "Breland's",
        "Brelish",
        "Breven",
        "Brokenblade",
        "bulette",
        "bulettes",
        "Caden",
        "Caden's",
        "Callestan",
        "Cannith",
        "cannith",
        "Cannith's",
        "Chervina",
        "Cliffside",
        "Cogsgate",
        "Creatlach",
        "Cyran",
        "Cyrans",
        "Cyre",
        "d'Cannith",
        "d'Deneith",
        "d'Ghallanda",
        "d'Jorasco",
        "d'Kundarak",
        "d'Lyrandar",
        "d'Medani",
        "d'Orien",
        "d'Phiarlan",
        "d'Sivis",
        "d'Tharashk",
        "d'Thuranni",
        "d'Vadalis",
        "d'Vol",
        "Daask",
        "Daask's",
        "daelkyr",
        "Daggerwatch",
        "Dalannan",
        "Dalin",
        "Darguul",
        "Darguun",
        "Darkvision",
        "Darna",
        "Deneith",
        "densewood",
        "dhakaan",
        "Dhakaani",
        "Dimas",
        "Dolom",
        "Dolurrh",
        "Dorasharn",
        "Dorn",
        "Dorn's",
        "Draal",
        "demihumans",
        "dragonmark",
        "dragonmarked",
        "dragonmarks",
        "dragonne",
        "Dragonshard",
        "dragonshards",
        "dreadhold",
        "dreamlily",
        "drik",
        "Droaam",
        "Droaamish",
        "Droaamites",
        "drow",
        "druidcraft",
        "Dura",
        "dwarven",
        "Eberron",
        "Edvin",
        "Elar",
        "Elaydren",
        "Elaydren's",
        "Eldeen",
        "Elorrenthi",
        "Eltarra",
        "Elvinor",
        "Erandis",
        "Eranna",
        "Esravash",
        "everbright",
        "extradimensional",
        "Failin",
        "Failin's",
        "Fairhaven",
        "Feru",
        "feywild",
        "Foalswood",
        "forgeholds",
        "Galifar",
        "Garm",
        "Garra",
        "Garra's",
        "Gatherhold",
        "Geldem",
        "gemwright",
        "Germaine",
        "Ghaal",
        "Ghaal'dar",
        "Ghallanda",
        "Ghallanda's",
        "Ghalt",
        "glamerweave",
        "Gnoll",
        "goblinoid",
        "goblinoids",
        "Gomba",
        "Griby",
        "Halamar",
        "Halas",
        "Halfling",
        "halflings",
        "Haruuc",
        "Haruuc's",
        "Heward",
        "Heward's",
        "Higglefritz",
        "hippogriffs",
        "Ilyra",
        "inquisitives",
        "ir'Morgrave",
        "ir'Talan",
        "ir'Wynarn",
        "Iyan",
        "Jarina",
        "Jorasco",
        "Jorasco's",
        "Jorlanna",
        "Kalarel",
        "kalashtar",
        "Karrlakton",
        "Karrnath",
        "Karrnathi",
        "Karrns",
        "Katra",
        "Kedran",
        "Kelsa",
        "Kenku",
        "Khaal",
        "Khaar",
        "Khandar",
        "Khandar'aashta",
        "Khoravar",
        "Khorvaire",
        "Kibbil",
        "Kor",
        "Korçë",
        "Korran",
        "Korranberg",
        "Korryn",
        "Korryn's",
        "Korth",
        "korunda",
        "Kruj",
        "Krujë",
        "Kundarak",
        "kundarak",
        "Kundarak's",
        "Kwanti",
        "Kwanti's",
        "lareth",
        "Larloch",
        "Larriman",
        "Lashtavar",
        "Lathleer",
        "Leylani",
        "Lhazaar",
        "Lhesh",
        "Lian",
        "lich",
        "lightbringer",
        "lightbringers",
        "lizardman",
        "Lyrandar",
        "Lyrriman",
        "Lysse",
        "Maagrim",
        "Mabar",
        "magebred",
        "magebreed",
        "magebreeding",
        "Magewright",
        "Magewright's",
        "magewrights",
        "Malleon's",
        "manticore",
        "Marguul",
        "Mbar'ost",
        "Medani",
        "Medani's",
        "Mekdall",
        "Menthis",
        "Merrix",
        "Methis",
        "Mithral",
        "Moghadam",
        "Mordai",
        "Morgrave",
        "morrikan",
        "Mournland",
        "Mror",
        "Muut",
        "Nairo",
        "Nairoladnam",
        "Neebur",
        "Northedge",
        "NPCs",
        "Oargev",
        "Octagram",
        "Olladra",
        "Olladra's",
        "Onatar",
        "Orien",
        "Orien's",
        "Phiarlan",
        "Phiarlan's",
        "Raincallers",
        "Redcloak",
        "Redcloaks",
        "Reeva",
        "Regalport",
        "Retha",
        "Rhest",
        "Rhukaan",
        "s",
        "Sagart",
        "sahuagin",
        "Sarlona",
        "Sarlonan",
        "sentira",
        "Seodra",
        "shadowfell",
        "Shae",
        "Sharaat'kor",
        "Shargon",
        "Sharn",
        "sharn",
        "Sharn's",
        "Sivis",
        "Sivis's",
        "Skakan",
        "Skakan's",
        "skybridge",
        "skycoaches",
        "skyships",
        "spellcraft",
        "Starilaskur",
        "Sterngate",
        "Stonestalkers",
        "Stormhome",
        "Stormhome's",
        "subsects",
        "Sundry",
        "Sundry's",
        "sunrods",
        "taggit",
        "Talenta",
        "Talentan",
        "Tarkanan",
        "Tavick",
        "Tavick's",
        "teleporter",
        "Thaliost",
        "Tharashk",
        "Tharask",
        "Thessalar",
        "Thrane",
        "Thranes",
        "Thronehold",
        "Thuranni",
        "Tiefling",
        "Toranak",
        "Torrn",
        "treants",
        "Trelib",
        "Ulara",
        "undeath",
        "Urfather",
        "Ursas",
        "Ursus",
        "Vadalis",
        "Valenar",
        "Valyre",
        "Vathirond",
        "Vedykar",
        "Velderan",
        "Vilroy",
        "Vilroy's",
        "Vishtai",
        "Vohilda",
        "Vol",
        "Vown",
        "wandslingers",
        "warforged",
        "waterskin",
        "weaponsmith",
        "Whitehearth",
        "wildshape",
        "Wroann's",
        "Wroat",
        "Wyld",
        "Wynarn",
        "Xen",
        "Xen'drik",
        "Yannah",
        "Yoren",
        "Zarash",
        "Zarash'ak",
        "Zilargo",
        "Zorlan"
    ]
}
```

</details>

**Comment by vip30** [comment]
I have the same issue even with only one entry in cspell.json

**Comment by Jason3S** [comment]
This should be fixed with version 1.10.2
